GeometryCore/src/Mesh.cpp

```diff
@@ -208,4 +208,57 @@ namespace geom_utils
 		}
 		return foundation;
 	}
+	Mesh integrateFoundationIntoModel(const Mesh& model, const Mesh& foundation)
```

Comment:
In general, it is workable solution (except that your function must return Mesh object, and what it will return if `(!model.isEmpty())` is false? Also, here `*std::min_element(getZFoundation.begin(),getZFoundation.end());` you will have crash of the program if input `foundation` is empty, because you will be dereferencing invalid iterator).

But all this is over-complicated, as for me.

This function can be simplified. Algorithm is simply and clearly. Note how the following implementation is emphasizing that:
```
Mesh result;
if (!model.empty() && !foundation.empty())
{
  auto zDiff = model.getAABB().min.z - foundation.getAABB().max.z;
  zDiff -= gap; // apply additional gap
  foundation.translate(FPoint(0, 0, zDiff));
  result.merge(model);
  result.merge(foundation);
}
return result;
```
So, you just should provide implementation for `Mesh::getAABB()`, `Mesh::translate()`, `Mesh::merge()` methods. These methods are pretty straightforward. In fact, you has already implemented all these methods, but just dumped all on a heap here in one place.